app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,119 @@
+ *
+ * @property defaultValue The default value of this preference.
+ * @property min The minimum value of the seek bar. Must not be greater or equal than [max].
+ * @property max The maximum value of the seek bar. Must not be lesser or equal than [min].
+ * @property step The step in which the seek bar increases per move. If the provided value is less
+ *  than 1, 1 will be used as step.
+ * @property unit The unit to show after the value. Set to an empty string to disable this feature.
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    private var step: Int = 1
+    private var unit: String = ""
+
+    @Suppress("unused")
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) :
+            this(context, attrs, R.attr.preferenceStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        onPreferenceChangeListener = OnPreferenceChangeListener { _, newValue ->
+            summary = getTextForValue(newValue.toString())
+            true
+        }
+        onPreferenceClickListener = OnPreferenceClickListener {
+            showSeekBarDialog()
+            true
+        }
+    }
+
+    override fun onAttachedToHierarchy(preferenceManager: PreferenceManager?) {
+        super.onAttachedToHierarchy(preferenceManager)
+        summary = getTextForValue(
+            preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+        )
+    }
+
+    /**
+     * Generates the text for the given [value] and adds the defined [unit] at the end.
+     */
+    private fun getTextForValue(value: Any): String {
+        return if (value !is Int) {
+            "??$unit"
+        } else {
+            value.toString() + unit
+        }
+    }
+
+    /**
+     * Shows the seek bar dialog.
+     */
+    private fun showSeekBarDialog() {
+        val dialogView = DialogSeekBarBinding.inflate(LayoutInflater.from(context))
+        val initValue = preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+        dialogView.seekBar.max = actualValueToSeekBarProgress(max)
+        dialogView.seekBar.progress = actualValueToSeekBarProgress(initValue)
+        dialogView.seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                dialogView.seekBarValue.text = getTextForValue(seekBarProgressToActualValue(progress))
+            }
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {}
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {}
+        })
+        dialogView.seekBarValue.text = getTextForValue(initValue)
+        AlertDialog.Builder(context).apply {
+            setTitle(this@DialogSeekBarPreference.title)
+            setCancelable(true)
```

Comment:
Dialogs are cancelable by default iirc.

Reply:
Let us call it explicitly since default behavior may be different among several Android version.

Reply:
https://developer.android.com/reference/androidx/appcompat/app/AlertDialog.Builder#setCancelable(boolean)

> Sets whether the dialog is cancelable or not. Default is true.

`setCancelable` is considered public API and it's behavior shouldn't vary among different Android versions.